actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1264,8 +1278,16 @@ def resources(*resources, &block)
             concerns(options[:concerns]) if options[:concerns]
 
             collection do
-              get  :index if parent_resource.actions.include?(:index)
-              post :create if parent_resource.actions.include?(:create)
+              actions = parent_resource.actions
+              (options[:constraints] ||= {}).merge! parent_resource.collection_param => COLLECTION_PARAM_REGEXP if parent_resource.collection_routing?
+              get    :index, options if actions.include?(:index)
+              post   :create, options if actions.include?(:create)
+              if parent_resource.collection_routing?
+                put    :replace, options if actions.include?(:replace)
+                patch  :update_many, options if actions.include?(:update_many)
+                delete :destroy_many, options if actions.include?(:destroy_many)
+                get    :edit_many, options.merge(as: 'edit') if actions.include?(:edit_many)
+              end
             end
 
             new do
```

Comment:
Similarly, you might have to move the following `set_member_mappings_for_resource` **above** the `collection` block.

If you don't do that, a route like `/posts/1` will be processed as an index action where "1" is the "ids" parameter:

```sh
Started GET "/posts/1" for ::1 at 2015-01-12 12:10:56 -0800
Processing by PostsController#index as HTML
  Parameters: {"collection"=>true, "ids"=>"1"}
Unpermitted parameter: collection
Completed 404 Not Found in 1ms (ActiveRecord: 0.1ms)
```

If you move the block above `collection`, then you get the expected result:

```sh
Started GET "/posts/1" for ::1 at 2015-01-12 12:11:08 -0800
Processing by PostsController#show as HTML
  Parameters: {"id"=>"1"}
  Post Load (0.1ms)  SELECT  "posts".* FROM "posts" WHERE "posts"."id" = ? LIMIT 1  [["id", 1]]
  Rendered posts/show.html.erb within layouts/application (1.4ms)
Completed 200 OK in 20ms (Views: 7.3ms | ActiveRecord: 0.4ms)
```